proposals/0002-gep.md

```diff
@@ -0,0 +1,216 @@
+This request schema matches the OpenAI Chat request schema. The message field is required and the messageHistory field is optional. The role and content fields must be
+specified. These fields will be parsed when a specific provider is selected and messages will be passed in the correct schema to that provider.
+
+Unified API schema example:
+
+```json
+{
+  "message": [
+      {
+        "role": "user",
+        "content": "Where was it played?"
+      }
+    ],
+  "messageHistory": [
+    {"role": "system", "content": "You are a helpful assistant."},
+    {"role": "user", "content": "Who won the world series in 2020?"},
+    {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."},
+  ]
+}
+```
+If a user wants to specify different prompts for different models, they can do so.
+This assumes the following models have been added to the gateway via the config file.
+
+```json
+{
+  "model": {
+    "name": "gpt-3.5-turbo",
+    "messages": [
+      {
+        "role": "user",
+        "content": "Where was it played?"
+      }
+    ],
+    "messageHistory": [
+      {"role": "system", "content": "You are a helpful assistant."},
+      {"role": "user", "content": "Who won the world series in 2020?"},
+      {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."}
+    ]
+  },
+  "model": {
+    "name": "command-light",
+    "messages": [
+      {
+        "role": "user",
+        "content": "Answer this question in two words: Where was it played?"
+      }
+    ],
+    "messageHistory": [
+      {"role": "system", "content": "You are a helpful assistant."},
+      {"role": "user", "content": "Who won the world series in 2020?"},
+      {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."}
+    ]
+  }
+}
+
+```
+
+### Provider Request Schemas
+
+OpenAi Chat Endpoint Request Schema
+
+```json
+
+{
+    "model": "gpt-3.5-turbo",
+    "messages": [
+    {"role": "system", "content": "You are a helpful assistant."},
+    {"role": "user", "content": "Who won the world series in 2020?"},
+    {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."},
+    {"role": "user", "content": "Where was it played?"},
+    ]
+}
+
+```
+Cohere Chat Endpoint Request Schema
+
+```json
+{
+    "model": "command-light",
+    "chat_history": [
+      {"role": "USER", "message": "Who won the world series in 2020?"},
+      {"role": "CHATBOT", "message": "The Los Angeles Dodgers won the World Series in 2020."}
+    ],
+    "message": "Where was it played?"
+}
+```
+
+### Response Schema For Chat
+
+For MVP, lets just return the response from the provider. Once we get feedback we can incorporate more metedata from the gateway about the request.
+
+Unified Response Schema from OpenAI API to user
+
+```json
+{
+  "provider": "openai",
```

Comment:
I think this response schema is great. We return information about what model/provider ended up service the request and this must be super useful during troubleshooting, etc.